Build "Save and add another" secondary submit in hardware forms.

Caleb-Ellis opened the pull request:
## Done
- Add secondary submit props to FormCardButtons to allow another way to submit a form.
- In this case the the secondary submit is the same as the normal one, but the default "redirect on save" behaviour is prevented.

## QA
- Go to `/MAAS/r/machines/add` and fill the form in correctly.
- Click "Save and add another" and check that the new machine is added into state, a notification confirms it, and you are not redirected to the machine list.
- Fill in the form again and click "Save machine".
- Check that you are redirected to the machine list, as before.
- Do the same with the "Add chassis" form (`/MAAS/r/machines/chassis/add`)

## Fixes
Fixes #795 

## Screenshot

![0 0 0 0_8400_MAAS_r_machines (17)](https://user-images.githubusercontent.com/25733845/75273983-358f8b00-5802-11ea-9fb3-aad957ef6612.png)

squidsoup commented on ui/src/app/base/components/FormCardButtons/FormCardButtons.js:
```diff
@@ -20,6 +24,18 @@ export const FormCardButtons = ({
+        <Button
+          appearance="neutral"
+          className="u-no-margin--bottom"
+          data-test="secondary-submit"
```
@huwshimi I think they're okay in production code, they affect performance or alter behaviour. In fact I think we probably *want* them in the production build for e2e tests.

Caleb-Ellis replied:
agreed with @squidsoup that it'll be useful for e2e tests, but it's actually already useful because it makes these unit tests less brittle. Usually their purpose isn't to test that the DOM has a certain shape, so why bind a test to the DOM shape when all you care about is a single button?

squidsoup commented on ui/src/app/base/components/FormCardButtons/FormCardButtons.js:
```diff
@@ -1,15 +1,19 @@
   actionLoading,
-  actionSuccess
+  actionSuccess,
+  handleSecondarySubmit,
```
I feel like it is a bit weird that we have two handlers, with quite different names, `actionSuccess` and `handleSecondarySubmit`.

Caleb-Ellis replied:
I agree, but then again this whole pattern is a bit weird. `handleSecondarySubmit` is actually the only submit handler that this component knows about. The normal submit is handled by the parent form element (via the `type="submit"` button) and `actionSuccess` isn't actually a submit handler, it's just to show the success icon in the ActionButton. I'll try and come up with some clearer names.

squidsoup replied:
Ah I see, that's quite fine in that case, sorry for the confusion. Thanks @Caleb-Ellis

Caleb-Ellis commented on ui/src/app/base/components/FormCardButtons/FormCardButtons.js:
```diff
@@ -20,6 +24,18 @@ export const FormCardButtons = ({
+          className="u-no-margin--bottom"
+          data-test="secondary-submit"
+          disabled={actionDisabled}
+          onClick={() => handleSecondarySubmit()}
```
make sense :)

Caleb-Ellis commented on ui/src/app/base/components/FormikFormContent/FormikFormContent.js:
```diff
@@ -47,6 +49,11 @@ const FormikFormContent = ({
         actionLabel={submitLabel}
         actionLoading={saving}
         actionSuccess={saved}
+        handleSecondarySubmit={() => {
```
yes, good catch

Caleb-Ellis commented:
@huwshimi Thanks for the review! I've updated some of the names for props which hopefully make it a bit clearer what's going on. As for the reset form I implemented a simple version using formik's `resetForm` prop on submit, but I noticed that if the api returns an error the form is cleared anyway and it's annoying to have to re-enter the form fields. I've [added a card](https://github.com/canonical-web-and-design/maas-ui/issues/844) to clear the form on *successful* form submit which I'll tackle separately.
